### PR TITLE
[MRG] Enable moltypes other than DNA in LCA databases

### LINKED ISSUE #905: sourmash signature describe misses dayhoff, hp molecule types
Looks like signature describe isn't handling dayhoff, hp encoding properly. In the example below, these dayhoff & hp sigs are described as DNA sigs. 

protein signature example, with protein, dayhoff, hp molecule encodings: `test.sig`:  

```
[{"class":"sourmash_signature","email":"","hash_function":"0.murmur64","filename":"test.prot","license":"CC0","signatures":[{"num":0,"ksize":11,"seed":42,"max_hash":922337203685477632,"mins":                     [8654912015719540,36111891283354331,37308772364955832,87386056322424681,172394428206145264,175324458129672500,192458836906930061,203682007118818914,228487149606072746,249673742953190691,252759730738350599,       269851384402800409,276879657160814740,350146213620382349,355128279925181821,362142414738517052,372280419647394145,378568520451171994,400417693755427902,411540066123110581,447785525058093308,                      508503262931135755,558153676748293637,603380202800315060,608284974085382699,672722089804872372,703845239558431174,717802877765604512,725736535145543392,773956917118790885,802909586863648861,                      874660063252901754,877868933880276449,891726720659491306],"md5sum":"57ae47c24a08a24b630dae80f0c6e256","abundances":[1,2,2,1,2,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,2,1,1,1,1,2,1,1,1,1,1],"molecule":"protein"},     {"num":0,"ksize":11,"seed":42,"max_hash":922337203685477632,"mins":[394785509353664721,461721674775207677,709435529205169740,851683204813363936],"md5sum":"407a774f7a2e13c650ff67255d2056f8","abundances":[2,4,     1,10],"molecule":"dayhoff"},{"num":0,"ksize":11,"seed":42,"max_hash":922337203685477632,"mins":[48718379107742671,311269990247623707],"md5sum":"72560192a7f4c23f0b4d71b3eaaa44db","abundances":[112,89], "molecule":"hp"}],"version":0.4}]
```

Running `sourmash signature describe` on this:

```
$ sourmash signature describe test.sig

== This is sourmash version 3.2.2. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

loaded 3 signatures total.
---
signature filename: test.sig
signature: test.prot
source file: test.prot
md5: 57ae47c24a08a24b630dae80f0c6e256
k=11 molecule=protein num=0 scaled=20 seed=42 track_abundance=1
size: 34
signature license: CC0

---
signature filename: test.sig
signature: test.prot
source file: test.prot
md5: 407a774f7a2e13c650ff67255d2056f8
k=11 molecule=DNA num=0 scaled=20 seed=42 track_abundance=1
size: 4
signature license: CC0

---
signature filename: test.sig
signature: test.prot
source file: test.prot
md5: 72560192a7f4c23f0b4d71b3eaaa44db
k=11 molecule=DNA num=0 scaled=20 seed=42 track_abundance=1
size: 2
signature license: CC0
```

### LINKED ISSUE #947: support protein LCA database?
with the LCA Database refactor in #946, it turns out to be pretty easy to support protein LCA databases - just need to change one line! (well, and support that option in creating, loading, and saving). See comment in function `LCA_Database._signatures(...)` for more information.

ref #637

### LINKED ISSUE #1012: sourmash lca index doesn't give good error messages for diff ksize, non-DNA, etc

### LINKED ISSUE #1014: sourmash sig describe has broken moltype output
UPDATE: sourmash sig describe had broken moltype output.

```
sourmash compute -k 19 ../*_protein.faa.gz --scaled=100 -f --input-is-protein  --hp --no-protein
```
produces DNA and protein signatures, but no hp signatures.

### PR DESCRIPTION
**Pull request by ctb**
This PR adds support for protein, dayhoff, and hp signatures to `LCA_Database` and `sourmash lca index`. Things now work as expected on the command line, although the default scaled and ksize for `sourmash lca index` don't really make sense and we might want to change them in some future version.

Fixes #947 

Also:
* Bumps LCA database version to 2.1.
* Also adds more/better tests for SBT index and search/gather of protein, dayhoff, and hp signatures.
* Fixes #1012 poor error reporting
* Fixes #905 sourmash sig describe yields bad moltype
* Fixes #1014 dup #905 
* improves reporting of sourmash lca index in various ways ref https://github.com/dib-lab/sourmash/issues/1000#issuecomment-636832003.
* fixes minor reporting bug with remnant_records in command_index, adds related lineage checking assert
* supercedes #907 

Tests to write:
- [x] sig describe moltype stuff
- [x]  LCA_Database insert with wrong moltype
- [x] LCA_Database insert with return val

----

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
@bluegenes could you review this, please? thanks!

**Comment by ctb** [comment]
Yay! Tests passed!

File: tests/test_lca.py [hunk 1]
```diff
@@ -1834,3 +1848,330 @@ def test_gather_multiple_return(c):
...
+
+def test_lca_db_protein_build():
+    # test programmatic creation of LCA database with protein sigs in it
+    sigfile1 = utils.get_test_data('prot/protein/GCA_001593925.1_ASM159392v1_protein.faa.gz.sig')
+    sigfile2 = utils.get_test_data('prot/protein/GCA_001593935.1_ASM159393v1_protein.faa.gz.sig')
+
+    sig1 = sourmash.load_one_signature(sigfile1)
+    sig2 = sourmash.load_one_signature(sigfile2)
+    
+    db = sourmash.lca.LCA_Database(ksize=19, scaled=100, moltype='protein')
```
Comment (bluegenes) [inline tests/test_lca.py:1861]: for soumash 3.x, protein kmers should be divisible by 3. Here, 19/3 = 6.3, so the actual ksize being used is k= 6.

We should maybe reimplement the check that #575/ #576 removed?

**Comment by ctb** [comment]
thanks @bluegenes! I changed the ksize=19 to ksize=57 in 20d5cf5 and 59a6fb8 - will merge if / when the tests pass!

**Pull request merged**